<html lang="en">
<head>
    <link href="http://getbootstrap.com/dist/css/bootstrap.min.css" rel="stylesheet">
	<style type="text/css">
		pre {
			width:1200px;
			overflow: auto;
			word-wrap: normal;
			white-space: pre;
		}
	</style>
	<script src="http://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js"></script>
	<script src="http://cytoscape.github.io/cytoscape.js/api/cytoscape.js-latest/cytoscape.min.js"></script>
	<script src="Submit_cri_rnaseq_2018.bds.20181114_103932_986.dag.js"></script>
</head>
<body>

<div class="container" id="summary">
	<h2 class="bs-featurette-title"> BigDataScript report: Submit_cri_rnaseq_2018.bds </h2>

	<table class="table table-striped">
		<tr> <th> Script file </th><td> /CRI/HPC/cri_rnaseq_2018/Submit_cri_rnaseq_2018.bds </td></tr>
		<tr> <th> Program ID </th><td> Submit_cri_rnaseq_2018.bds.20181114_103932_986 </td></tr>
		<tr> <th> Start time </th><td> 2018-11-14 10:39:32 </td></tr>
		<tr> <th> Run time </th><td> 00:27:33.131 </td></tr>
		<tr> <th> Tasks executed </th><td> 49 </td></tr>
		<tr> <th> Tasks failed </th><td> 0 </td></tr>
		<tr> <th> Tasks failed names </th><td> <pre></pre> </td></tr>
		<tr> <th> Arguments* </th><td> [] </td></tr>
		<tr> <th> System* </th><td> cluster </td></tr>
		<tr> <th> Cpus* </th><td> 1 </td></tr>
		<tr> <th> Exit value </th><td  > 0 </td></tr>
	</table>
	<b>*</b> Values in global scope when program finished execution.
</div>

<div class="container" id="bdsTimeLine">
	<h2 class="bs-featurette-title"> Timeline </h2>
	<script type="text/javascript" src="https://www.google.com/jsapi?autoload={'modules':[{'name':'visualization','version':'1','packages':['timeline']}]}"></script>
	<script type="text/javascript">
	google.setOnLoadCallback(drawChart);

	function drawChart() {
	  var container = document.getElementById('bdsTimeline');

	  var chart = new google.visualization.Timeline(container);

	  var dataTable = new google.visualization.DataTable();

	  dataTable.addColumn({ type: 'string', id: 'TaskName' });
	  dataTable.addColumn({ type: 'string', id: 'TaskHint' });
	  dataTable.addColumn({ type: 'date', id: 'Start' });
	  dataTable.addColumn({ type: 'date', id: 'End' });

	  dataTable.addRows([
	    [ 'FastQC.SRR1205282', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205282.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,45,30) ],
	    [ 'FastQC.SRR1205283', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205283.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,44,47) ],
	    [ 'FastQC.SRR1205284', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205284.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,43,51) ],
	    [ 'FastQC.SRR1205285', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205285.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,44,00) ],
	    [ 'FastQC.SRR1205286', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205286.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,44,23) ],
	    [ 'FastQC.SRR1205287', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205287.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,44,55) ],
	    [ 'star.SRR1205282', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205282.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,51,24) ],
	    [ 'star.SRR1205283', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205283.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,49,57) ],
	    [ 'star.SRR1205284', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205284.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,48,07) ],
	    [ 'star.SRR1205285', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205285.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,48,28) ],
	    [ 'star.SRR1205286', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205286.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,49,14) ],
	    [ 'star.SRR1205287', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205287.sh; sleep 2', new Date(2018,10,14,10,39,33), new Date(2018,10,14,10,49,51) ],
	    [ 'star.KO01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.KO01.sh; sleep 2', new Date(2018,10,14,10,51,24), new Date(2018,10,14,10,52,34) ],
	    [ 'star.KO02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.KO02.sh; sleep 2', new Date(2018,10,14,10,49,57), new Date(2018,10,14,10,50,51) ],
	    [ 'star.KO03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.KO03.sh; sleep 2', new Date(2018,10,14,10,48,07), new Date(2018,10,14,10,48,59) ],
	    [ 'star.WT01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.WT01.sh; sleep 2', new Date(2018,10,14,10,48,28), new Date(2018,10,14,10,50,44) ],
	    [ 'star.WT02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.WT02.sh; sleep 2', new Date(2018,10,14,10,49,14), new Date(2018,10,14,10,50,48) ],
	    [ 'star.WT03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.WT03.sh; sleep 2', new Date(2018,10,14,10,49,51), new Date(2018,10,14,10,50,50) ],
	    [ 'picard.star.KO01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.KO01.CollectRnaSeqMetrics.sh; sleep 2', new Date(2018,10,14,10,52,34), new Date(2018,10,14,11,07,06) ],
	    [ 'picard.star.KO02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.KO02.CollectRnaSeqMetrics.sh; sleep 2', new Date(2018,10,14,10,50,51), new Date(2018,10,14,11,03,07) ],
	    [ 'picard.star.KO03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.KO03.CollectRnaSeqMetrics.sh; sleep 2', new Date(2018,10,14,10,48,59), new Date(2018,10,14,10,59,33) ],
	    [ 'picard.star.WT01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.WT01.CollectRnaSeqMetrics.sh; sleep 2', new Date(2018,10,14,10,50,44), new Date(2018,10,14,11,02,15) ],
	    [ 'picard.star.WT02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.WT02.CollectRnaSeqMetrics.sh; sleep 2', new Date(2018,10,14,10,50,48), new Date(2018,10,14,11,02,52) ],
	    [ 'picard.star.WT03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.WT03.CollectRnaSeqMetrics.sh; sleep 2', new Date(2018,10,14,10,50,50), new Date(2018,10,14,11,03,44) ],
	    [ 'rseqc.star.KO01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO01.clipping_profile.py.sh; sleep 2', new Date(2018,10,14,10,52,34), new Date(2018,10,14,10,57,34) ],
	    [ 'rseqc.star.KO01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO01.infer_experiment.py.sh; sleep 2', new Date(2018,10,14,10,52,34), new Date(2018,10,14,10,53,21) ],
	    [ 'rseqc.star.KO02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO02.clipping_profile.py.sh; sleep 2', new Date(2018,10,14,10,50,51), new Date(2018,10,14,10,55,18) ],
	    [ 'rseqc.star.KO02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO02.infer_experiment.py.sh; sleep 2', new Date(2018,10,14,10,50,51), new Date(2018,10,14,10,53,23) ],
	    [ 'rseqc.star.KO03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO03.clipping_profile.py.sh; sleep 2', new Date(2018,10,14,10,48,59), new Date(2018,10,14,10,53,03) ],
	    [ 'rseqc.star.KO03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO03.infer_experiment.py.sh; sleep 2', new Date(2018,10,14,10,48,59), new Date(2018,10,14,10,50,13) ],
	    [ 'rseqc.star.WT01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT01.clipping_profile.py.sh; sleep 2', new Date(2018,10,14,10,50,44), new Date(2018,10,14,10,54,47) ],
	    [ 'rseqc.star.WT01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT01.infer_experiment.py.sh; sleep 2', new Date(2018,10,14,10,50,44), new Date(2018,10,14,10,51,46) ],
	    [ 'rseqc.star.WT02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT02.clipping_profile.py.sh; sleep 2', new Date(2018,10,14,10,50,48), new Date(2018,10,14,10,55,11) ],
	    [ 'rseqc.star.WT02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT02.infer_experiment.py.sh; sleep 2', new Date(2018,10,14,10,50,48), new Date(2018,10,14,10,51,46) ],
	    [ 'rseqc.star.WT03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT03.clipping_profile.py.sh; sleep 2', new Date(2018,10,14,10,50,50), new Date(2018,10,14,10,55,27) ],
	    [ 'rseqc.star.WT03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT03.infer_experiment.py.sh; sleep 2', new Date(2018,10,14,10,50,50), new Date(2018,10,14,10,51,47) ],
	    [ 'featurecounts.star.KO01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.KO01.sh; sleep 2', new Date(2018,10,14,10,52,34), new Date(2018,10,14,10,55,56) ],
	    [ 'featurecounts.star.KO02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.KO02.sh; sleep 2', new Date(2018,10,14,10,50,51), new Date(2018,10,14,10,54,01) ],
	    [ 'featurecounts.star.KO03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.KO03.sh; sleep 2', new Date(2018,10,14,10,48,59), new Date(2018,10,14,10,52,00) ],
	    [ 'featurecounts.star.WT01', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.WT01.sh; sleep 2', new Date(2018,10,14,10,50,44), new Date(2018,10,14,10,53,42) ],
	    [ 'featurecounts.star.WT02', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.WT02.sh; sleep 2', new Date(2018,10,14,10,50,48), new Date(2018,10,14,10,53,53) ],
	    [ 'featurecounts.star.WT03', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.WT03.sh; sleep 2', new Date(2018,10,14,10,50,50), new Date(2018,10,14,10,54,02) ],
	    [ 'edger.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.call.edger.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,55,56), new Date(2018,10,14,10,57,01) ],
	    [ 'deseq2.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.call.deseq2.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,55,57), new Date(2018,10,14,10,57,21) ],
	    [ 'limma.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.call.limma.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,55,57), new Date(2018,10,14,10,57,01) ],
	    [ 'pca.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quantQC.pca.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,57,21), new Date(2018,10,14,10,58,21) ],
	    [ 'lociStat.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.lociStat.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,57,21), new Date(2018,10,14,10,58,22) ],
	    [ 'postAna.pheatmap.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.postAna.pheatmap.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,58,22), new Date(2018,10,14,11,00,02) ],
	    [ 'postAna.clusterprofiler.featurecounts.star.cri_rnaseq_2018', 'bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.postAna.clusterprofiler.featurecounts.star.cri_rnaseq_2018.sh; sleep 2', new Date(2018,10,14,10,58,22), new Date(2018,10,14,11,03,18) ],
	    ]);

	  var options = {
	    timeline: { groupByRowLabel: false }
	  };

	  chart.draw(dataTable, options);
	}
	</script>
	<div id="bdsTimeline" style="width: 900px; height: 2142px;"></div>
</div>

<div class="container" id="taskDag">
	<h2 class="bs-featurette-title"> Task Graph </h2>
	<div class="container" id="cy" style="height: 100%; width: 100%; border: solid 1px black""></div>
</div>

<div class="container" id="threadDetails">
	<h2 class="bs-featurette-title"> Parallel (threads) details </h2>
	<table class="table table-striped">
    <tr> <th> Thread ID    </th> <th> Parent           </th>	<th> Tasks </th> </tr>
	<tr> <td> thread_Root </td> <td> None </td>	<td> <pre> Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.FastQC.SRR1205282.line_9.id_1
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.FastQC.SRR1205283.line_15.id_2
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.FastQC.SRR1205284.line_21.id_3
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.FastQC.SRR1205285.line_27.id_4
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.FastQC.SRR1205286.line_33.id_5
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.FastQC.SRR1205287.line_39.id_6
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.SRR1205282.line_53.id_7
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.SRR1205283.line_59.id_8
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.SRR1205284.line_65.id_9
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.SRR1205285.line_71.id_10
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.SRR1205286.line_77.id_11
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.SRR1205287.line_83.id_12
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.KO01.line_97.id_13
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.KO02.line_104.id_14
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.KO03.line_111.id_15
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.WT01.line_118.id_16
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.WT02.line_125.id_17
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.star.WT03.line_132.id_18
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.picard.star.KO01.line_146.id_19
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.picard.star.KO02.line_152.id_20
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.picard.star.KO03.line_158.id_21
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.picard.star.WT01.line_164.id_22
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.picard.star.WT02.line_170.id_23
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.picard.star.WT03.line_176.id_24
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.KO01.line_182.id_25
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.KO01.line_188.id_26
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.KO02.line_194.id_27
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.KO02.line_200.id_28
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.KO03.line_206.id_29
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.KO03.line_212.id_30
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.WT01.line_218.id_31
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.WT01.line_224.id_32
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.WT02.line_230.id_33
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.WT02.line_236.id_34
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.WT03.line_242.id_35
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.rseqc.star.WT03.line_248.id_36
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.featurecounts.star.KO01.line_262.id_37
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.featurecounts.star.KO02.line_268.id_38
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.featurecounts.star.KO03.line_274.id_39
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.featurecounts.star.WT01.line_280.id_40
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.featurecounts.star.WT02.line_286.id_41
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.featurecounts.star.WT03.line_292.id_42
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.edger.featurecounts.star.cri_rnaseq_2018.line_306.id_43
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.deseq2.featurecounts.star.cri_rnaseq_2018.line_312.id_44
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.limma.featurecounts.star.cri_rnaseq_2018.line_318.id_45
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.pca.featurecounts.star.cri_rnaseq_2018.line_332.id_46
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.lociStat.featurecounts.star.cri_rnaseq_2018.line_346.id_47
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.postAna.pheatmap.featurecounts.star.cri_rnaseq_2018.line_360.id_48
Submit_cri_rnaseq_2018.bds.20181114_103932_986/task.Submit_cri_rnaseq_2018.postAna.clusterprofiler.featurecounts.star.cri_rnaseq_2018.line_366.id_49
 </pre> </td> </tr>
	</table>
</div>

<div class="container" id="taskDetails">
	<h2 class="bs-featurette-title"> Task details </h2>
	<table class="table table-striped">
    <tr> <th> Task </th>            <th> Execution </th>            <th> Time </th> <th> Dependencies </th>     <th> Task program, Errors, StdOut / StdErr </th> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 1 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.FastQC.SRR1205282.line_9.id_1 </td> </tr> <tr> <th> Name </th> <td> FastQC.SRR1205282 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817683.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  1 </td></tr> <tr> <th> Mem </th> <td> 16.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:45:30 </td> </tr> <tr> <th> Elapsed </th> <td> 00:05:57 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205282.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/RawReadQC/KO01/SRR1205282/SRR1205282_fastqc.zip
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205282.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 2 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.FastQC.SRR1205283.line_15.id_2 </td> </tr> <tr> <th> Name </th> <td> FastQC.SRR1205283 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817684.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  1 </td></tr> <tr> <th> Mem </th> <td> 16.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:44:47 </td> </tr> <tr> <th> Elapsed </th> <td> 00:05:14 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205283.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/RawReadQC/KO02/SRR1205283/SRR1205283_fastqc.zip
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205283.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 3 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.FastQC.SRR1205284.line_21.id_3 </td> </tr> <tr> <th> Name </th> <td> FastQC.SRR1205284 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817685.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  1 </td></tr> <tr> <th> Mem </th> <td> 16.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:43:51 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:18 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205284.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/RawReadQC/KO03/SRR1205284/SRR1205284_fastqc.zip
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205284.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 4 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.FastQC.SRR1205285.line_27.id_4 </td> </tr> <tr> <th> Name </th> <td> FastQC.SRR1205285 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817686.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  1 </td></tr> <tr> <th> Mem </th> <td> 16.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:44:00 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:27 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205285.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/RawReadQC/WT01/SRR1205285/SRR1205285_fastqc.zip
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205285.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 5 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.FastQC.SRR1205286.line_33.id_5 </td> </tr> <tr> <th> Name </th> <td> FastQC.SRR1205286 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817687.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  1 </td></tr> <tr> <th> Mem </th> <td> 16.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:44:23 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:49 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205286.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/RawReadQC/WT02/SRR1205286/SRR1205286_fastqc.zip
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205286.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 6 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.FastQC.SRR1205287.line_39.id_6 </td> </tr> <tr> <th> Name </th> <td> FastQC.SRR1205287 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817688.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  1 </td></tr> <tr> <th> Mem </th> <td> 16.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:44:55 </td> </tr> <tr> <th> Elapsed </th> <td> 00:05:22 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205287.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/RawReadQC/WT03/SRR1205287/SRR1205287_fastqc.zip
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.RawReadQC.FastQC.SRR1205287.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 7 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.SRR1205282.line_53.id_7 </td> </tr> <tr> <th> Name </th> <td> star.SRR1205282 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817689.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:51:24 </td> </tr> <tr> <th> Elapsed </th> <td> 00:11:51 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205282.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/SRR1205282/SRR1205282.star.bam
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205282.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 8 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.SRR1205283.line_59.id_8 </td> </tr> <tr> <th> Name </th> <td> star.SRR1205283 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817690.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:49:57 </td> </tr> <tr> <th> Elapsed </th> <td> 00:10:23 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205283.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/SRR1205283/SRR1205283.star.bam
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205283.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 9 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.SRR1205284.line_65.id_9 </td> </tr> <tr> <th> Name </th> <td> star.SRR1205284 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817691.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:48:07 </td> </tr> <tr> <th> Elapsed </th> <td> 00:08:33 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205284.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/SRR1205284/SRR1205284.star.bam
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205284.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 10 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.SRR1205285.line_71.id_10 </td> </tr> <tr> <th> Name </th> <td> star.SRR1205285 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817692.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:48:28 </td> </tr> <tr> <th> Elapsed </th> <td> 00:08:55 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205285.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/SRR1205285/SRR1205285.star.bam
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205285.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 11 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.SRR1205286.line_77.id_11 </td> </tr> <tr> <th> Name </th> <td> star.SRR1205286 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817693.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:49:14 </td> </tr> <tr> <th> Elapsed </th> <td> 00:09:41 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205286.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/SRR1205286/SRR1205286.star.bam
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205286.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 12 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.SRR1205287.line_83.id_12 </td> </tr> <tr> <th> Name </th> <td> star.SRR1205287 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817694.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:39:33 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:49:51 </td> </tr> <tr> <th> Elapsed </th> <td> 00:10:17 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/example/data/SRR1205287.fastq.gz
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/SRR1205287/SRR1205287.star.bam
 </td> </tr> <tr> <th> Dependencies </th> <td>  </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alignRead.star.SRR1205287.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 13 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.KO01.line_97.id_13 </td> </tr> <tr> <th> Name </th> <td> star.KO01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817809.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:51:24 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:52:34 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:10 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/SRR1205282/SRR1205282.star.bam
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/KO01.star.bai
 </td> </tr> <tr> <th> Dependencies </th> <td> star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
star.SRR1205282
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.KO01.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 14 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.KO02.line_104.id_14 </td> </tr> <tr> <th> Name </th> <td> star.KO02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817792.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:49:57 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:50:51 </td> </tr> <tr> <th> Elapsed </th> <td> 00:00:54 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/SRR1205283/SRR1205283.star.bam
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/KO02.star.bai
 </td> </tr> <tr> <th> Dependencies </th> <td> star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
star.SRR1205283
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.KO02.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 15 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.KO03.line_111.id_15 </td> </tr> <tr> <th> Name </th> <td> star.KO03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817695.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:48:07 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:48:59 </td> </tr> <tr> <th> Elapsed </th> <td> 00:00:52 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/SRR1205284/SRR1205284.star.bam
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/KO03.star.bai
 </td> </tr> <tr> <th> Dependencies </th> <td> star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
star.SRR1205284
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.KO03.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 16 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.WT01.line_118.id_16 </td> </tr> <tr> <th> Name </th> <td> star.WT01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817785.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:48:28 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:50:44 </td> </tr> <tr> <th> Elapsed </th> <td> 00:02:15 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/SRR1205285/SRR1205285.star.bam
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/WT01.star.bai
 </td> </tr> <tr> <th> Dependencies </th> <td> star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
star.SRR1205285
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.WT01.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 17 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.WT02.line_125.id_17 </td> </tr> <tr> <th> Name </th> <td> star.WT02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817790.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:49:14 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:50:48 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:33 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/SRR1205286/SRR1205286.star.bam
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/WT02.star.bai
 </td> </tr> <tr> <th> Dependencies </th> <td> star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
star.SRR1205286
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.WT02.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 18 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.star.WT03.line_132.id_18 </td> </tr> <tr> <th> Name </th> <td> star.WT03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817791.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:49:51 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:50:50 </td> </tr> <tr> <th> Elapsed </th> <td> 00:00:59 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/SRR1205287/SRR1205287.star.bam
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/WT03.star.bai
 </td> </tr> <tr> <th> Dependencies </th> <td> star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
star.SRR1205287
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.mergeAln.star.WT03.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 19 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.picard.star.KO01.line_146.id_19 </td> </tr> <tr> <th> Name </th> <td> picard.star.KO01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817810.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:52:34 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:07:06 </td> </tr> <tr> <th> Elapsed </th> <td> 00:14:31 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/KO01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/KO01/KO01.star.picard.RNA_Metrics
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/KO01/KO01.star.picard.RNA_Metrics.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO01
star.KO01
star.KO01
star.KO01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.KO01.CollectRnaSeqMetrics.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 20 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.picard.star.KO02.line_152.id_20 </td> </tr> <tr> <th> Name </th> <td> picard.star.KO02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817805.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:51 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:03:07 </td> </tr> <tr> <th> Elapsed </th> <td> 00:12:16 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/KO02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/KO02/KO02.star.picard.RNA_Metrics
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/KO02/KO02.star.picard.RNA_Metrics.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO02
star.KO02
star.KO02
star.KO02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.KO02.CollectRnaSeqMetrics.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 21 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.picard.star.KO03.line_158.id_21 </td> </tr> <tr> <th> Name </th> <td> picard.star.KO03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817786.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:48:59 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:59:33 </td> </tr> <tr> <th> Elapsed </th> <td> 00:10:33 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/KO03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/KO03/KO03.star.picard.RNA_Metrics
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/KO03/KO03.star.picard.RNA_Metrics.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO03
star.KO03
star.KO03
star.KO03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.KO03.CollectRnaSeqMetrics.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 22 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.picard.star.WT01.line_164.id_22 </td> </tr> <tr> <th> Name </th> <td> picard.star.WT01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817793.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:44 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:02:15 </td> </tr> <tr> <th> Elapsed </th> <td> 00:11:31 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/WT01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/WT01/WT01.star.picard.RNA_Metrics
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/WT01/WT01.star.picard.RNA_Metrics.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT01
star.WT01
star.WT01
star.WT01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.WT01.CollectRnaSeqMetrics.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 23 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.picard.star.WT02.line_170.id_23 </td> </tr> <tr> <th> Name </th> <td> picard.star.WT02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817797.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:48 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:02:52 </td> </tr> <tr> <th> Elapsed </th> <td> 00:12:04 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/WT02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/WT02/WT02.star.picard.RNA_Metrics
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/WT02/WT02.star.picard.RNA_Metrics.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT02
star.WT02
star.WT02
star.WT02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.WT02.CollectRnaSeqMetrics.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 24 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.picard.star.WT03.line_176.id_24 </td> </tr> <tr> <th> Name </th> <td> picard.star.WT03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817801.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:50 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:03:44 </td> </tr> <tr> <th> Elapsed </th> <td> 00:12:53 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/WT03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/WT03/WT03.star.picard.RNA_Metrics
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/picard/star/WT03/WT03.star.picard.RNA_Metrics.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT03
star.WT03
star.WT03
star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.picard.star.WT03.CollectRnaSeqMetrics.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 25 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.KO01.line_182.id_25 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.KO01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817811.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:52:34 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:57:34 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:59 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/KO01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO01/KO01.star.rseqc.clipping_profile.xls
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO01/KO01.star.rseqc.clipping_profile.r
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO01/KO01.star.rseqc.clipping_profile.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO01.clipping_profile.py.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 26 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.KO01.line_188.id_26 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.KO01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817812.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:52:34 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:53:21 </td> </tr> <tr> <th> Elapsed </th> <td> 00:00:46 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/KO01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO01/KO01.star.rseqc.infer_experiment.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO01
star.KO01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO01.infer_experiment.py.sh; sleep 2 </pre> <pre>
--------------------Stderr--------------------
Reading reference gene model /gpfs/data/bioinformatics/cri_rnaseq_2018/example/references/v28_92_GRCh38.p12/genes.gtf.bed12 ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled

 </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 27 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.KO02.line_194.id_27 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.KO02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817806.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:51 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:55:18 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:26 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/KO02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO02/KO02.star.rseqc.clipping_profile.xls
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO02/KO02.star.rseqc.clipping_profile.r
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO02/KO02.star.rseqc.clipping_profile.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO02.clipping_profile.py.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 28 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.KO02.line_200.id_28 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.KO02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817807.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:51 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:53:23 </td> </tr> <tr> <th> Elapsed </th> <td> 00:02:32 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/KO02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO02/KO02.star.rseqc.infer_experiment.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO02
star.KO02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO02.infer_experiment.py.sh; sleep 2 </pre> <pre>
--------------------Stderr--------------------
Reading reference gene model /gpfs/data/bioinformatics/cri_rnaseq_2018/example/references/v28_92_GRCh38.p12/genes.gtf.bed12 ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled

 </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 29 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.KO03.line_206.id_29 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.KO03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817787.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:48:59 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:53:03 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:04 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/KO03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO03/KO03.star.rseqc.clipping_profile.xls
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO03/KO03.star.rseqc.clipping_profile.r
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO03/KO03.star.rseqc.clipping_profile.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO03.clipping_profile.py.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 30 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.KO03.line_212.id_30 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.KO03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817788.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:48:59 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:50:13 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:13 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/KO03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/KO03/KO03.star.rseqc.infer_experiment.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO03
star.KO03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.KO03.infer_experiment.py.sh; sleep 2 </pre> <pre>
--------------------Stderr--------------------
Reading reference gene model /gpfs/data/bioinformatics/cri_rnaseq_2018/example/references/v28_92_GRCh38.p12/genes.gtf.bed12 ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled

 </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 31 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.WT01.line_218.id_31 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.WT01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817794.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:44 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:54:47 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:03 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/WT01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT01/WT01.star.rseqc.clipping_profile.xls
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT01/WT01.star.rseqc.clipping_profile.r
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT01/WT01.star.rseqc.clipping_profile.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT01.clipping_profile.py.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 32 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.WT01.line_224.id_32 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.WT01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817795.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:44 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:51:46 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:02 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/WT01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT01/WT01.star.rseqc.infer_experiment.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT01
star.WT01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT01.infer_experiment.py.sh; sleep 2 </pre> <pre>
--------------------Stderr--------------------
Reading reference gene model /gpfs/data/bioinformatics/cri_rnaseq_2018/example/references/v28_92_GRCh38.p12/genes.gtf.bed12 ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled

 </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 33 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.WT02.line_230.id_33 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.WT02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817798.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:48 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:55:11 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:22 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/WT02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT02/WT02.star.rseqc.clipping_profile.xls
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT02/WT02.star.rseqc.clipping_profile.r
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT02/WT02.star.rseqc.clipping_profile.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT02.clipping_profile.py.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 34 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.WT02.line_236.id_34 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.WT02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817799.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:48 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:51:46 </td> </tr> <tr> <th> Elapsed </th> <td> 00:00:58 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/WT02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT02/WT02.star.rseqc.infer_experiment.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT02
star.WT02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT02.infer_experiment.py.sh; sleep 2 </pre> <pre>
--------------------Stderr--------------------
Reading reference gene model /gpfs/data/bioinformatics/cri_rnaseq_2018/example/references/v28_92_GRCh38.p12/genes.gtf.bed12 ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled

 </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 35 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.WT03.line_242.id_35 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.WT03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817802.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:50 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:55:27 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:36 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/WT03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT03/WT03.star.rseqc.clipping_profile.xls
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT03/WT03.star.rseqc.clipping_profile.r
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT03/WT03.star.rseqc.clipping_profile.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT03.clipping_profile.py.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 36 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.rseqc.star.WT03.line_248.id_36 </td> </tr> <tr> <th> Name </th> <td> rseqc.star.WT03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817803.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:50 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:51:47 </td> </tr> <tr> <th> Elapsed </th> <td> 00:00:56 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/WT03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/AlnQC/rseqc/star/WT03/WT03.star.rseqc.infer_experiment.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT03
star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.alnQC.rseqc.star.WT03.infer_experiment.py.sh; sleep 2 </pre> <pre>
--------------------Stderr--------------------
Reading reference gene model /gpfs/data/bioinformatics/cri_rnaseq_2018/example/references/v28_92_GRCh38.p12/genes.gtf.bed12 ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled

 </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 37 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.featurecounts.star.KO01.line_262.id_37 </td> </tr> <tr> <th> Name </th> <td> featurecounts.star.KO01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817813.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:52:34 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:55:56 </td> </tr> <tr> <th> Elapsed </th> <td> 00:03:22 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO01/KO01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO01/KO01.star.featurecounts.count
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
star.KO01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.KO01.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 38 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.featurecounts.star.KO02.line_268.id_38 </td> </tr> <tr> <th> Name </th> <td> featurecounts.star.KO02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817808.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:51 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:54:01 </td> </tr> <tr> <th> Elapsed </th> <td> 00:03:10 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO02/KO02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO02/KO02.star.featurecounts.count
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
star.KO02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.KO02.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 39 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.featurecounts.star.KO03.line_274.id_39 </td> </tr> <tr> <th> Name </th> <td> featurecounts.star.KO03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817789.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:48:59 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:52:00 </td> </tr> <tr> <th> Elapsed </th> <td> 00:03:01 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/KO03/KO03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO03/KO03.star.featurecounts.count
 </td> </tr> <tr> <th> Dependencies </th> <td> star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
star.KO03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.KO03.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 40 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.featurecounts.star.WT01.line_280.id_40 </td> </tr> <tr> <th> Name </th> <td> featurecounts.star.WT01 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817796.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:44 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:53:42 </td> </tr> <tr> <th> Elapsed </th> <td> 00:02:58 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT01/WT01.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT01/WT01.star.featurecounts.count
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
star.WT01
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.WT01.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 41 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.featurecounts.star.WT02.line_286.id_41 </td> </tr> <tr> <th> Name </th> <td> featurecounts.star.WT02 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817800.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:48 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:53:53 </td> </tr> <tr> <th> Elapsed </th> <td> 00:03:05 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT02/WT02.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT02/WT02.star.featurecounts.count
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
star.WT02
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.WT02.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 42 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.featurecounts.star.WT03.line_292.id_42 </td> </tr> <tr> <th> Name </th> <td> featurecounts.star.WT03 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817804.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:50:50 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:54:02 </td> </tr> <tr> <th> Elapsed </th> <td> 00:03:11 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Aln/star/WT03/WT03.star.bai
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT03/WT03.star.featurecounts.count
 </td> </tr> <tr> <th> Dependencies </th> <td> star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quant.featurecounts.star.WT03.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 43 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.edger.featurecounts.star.cri_rnaseq_2018.line_306.id_43 </td> </tr> <tr> <th> Name </th> <td> edger.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817814.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:55:56 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:57:01 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:04 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO01/KO01.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO02/KO02.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO03/KO03.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT01/WT01.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT02/WT02.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT03/WT03.star.featurecounts.count
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/edger/featurecounts/star/cri_rnaseq_2018.star.featurecounts.edger.count.txt
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/edger/featurecounts/star/cri_rnaseq_2018.star.featurecounts.edger.test.DEG.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.call.edger.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 44 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.deseq2.featurecounts.star.cri_rnaseq_2018.line_312.id_44 </td> </tr> <tr> <th> Name </th> <td> deseq2.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817815.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:55:57 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:57:21 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:24 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO01/KO01.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO02/KO02.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO03/KO03.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT01/WT01.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT02/WT02.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT03/WT03.star.featurecounts.count
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/deseq2/featurecounts/star/cri_rnaseq_2018.star.featurecounts.deseq2.count.txt
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/deseq2/featurecounts/star/cri_rnaseq_2018.star.featurecounts.deseq2.test.DEG.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.call.deseq2.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 45 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.limma.featurecounts.star.cri_rnaseq_2018.line_318.id_45 </td> </tr> <tr> <th> Name </th> <td> limma.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817816.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:55:57 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:57:01 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:04 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO01/KO01.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO02/KO02.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/KO03/KO03.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT01/WT01.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT02/WT02.star.featurecounts.count
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/Quantification/featurecounts/star/WT03/WT03.star.featurecounts.count
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/limma/featurecounts/star/cri_rnaseq_2018.star.featurecounts.limma.count.txt
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/limma/featurecounts/star/cri_rnaseq_2018.star.featurecounts.limma.test.DEG.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
featurecounts.star.KO01
featurecounts.star.KO02
featurecounts.star.KO03
featurecounts.star.WT01
featurecounts.star.WT02
featurecounts.star.WT03
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.call.limma.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 46 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.pca.featurecounts.star.cri_rnaseq_2018.line_332.id_46 </td> </tr> <tr> <th> Name </th> <td> pca.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817817.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  8 </td></tr> <tr> <th> Mem </th> <td> 64.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:57:21 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:58:21 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:00 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/deseq2/featurecounts/star/cri_rnaseq_2018.star.featurecounts.deseq2.count.txt
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/QuantQC/featurecounts/star/cri_rnaseq_2018.star.featurecounts.pca.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> deseq2.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.quantQC.pca.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 47 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.lociStat.featurecounts.star.cri_rnaseq_2018.line_346.id_47 </td> </tr> <tr> <th> Name </th> <td> lociStat.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817818.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:57:21 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 10:58:22 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:01 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/edger/featurecounts/star/cri_rnaseq_2018.star.featurecounts.edger.test.DEG.txt
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/deseq2/featurecounts/star/cri_rnaseq_2018.star.featurecounts.deseq2.test.DEG.txt
/gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/DEG/limma/featurecounts/star/cri_rnaseq_2018.star.featurecounts.limma.test.DEG.txt
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/LociStat/featurecounts/star/cri_rnaseq_2018/cri_rnaseq_2018.star.featurecounts.overlap.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> edger.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
limma.featurecounts.star.cri_rnaseq_2018
edger.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
limma.featurecounts.star.cri_rnaseq_2018
edger.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
limma.featurecounts.star.cri_rnaseq_2018
edger.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
limma.featurecounts.star.cri_rnaseq_2018
edger.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
limma.featurecounts.star.cri_rnaseq_2018
edger.featurecounts.star.cri_rnaseq_2018
deseq2.featurecounts.star.cri_rnaseq_2018
limma.featurecounts.star.cri_rnaseq_2018
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.lociStat.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 48 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.postAna.pheatmap.featurecounts.star.cri_rnaseq_2018.line_360.id_48 </td> </tr> <tr> <th> Name </th> <td> postAna.pheatmap.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817820.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:58:22 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:00:02 </td> </tr> <tr> <th> Elapsed </th> <td> 00:01:39 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/LociStat/featurecounts/star/cri_rnaseq_2018/cri_rnaseq_2018.star.featurecounts.overlap.txt
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/PostAna/pheatmap/featurecounts/star/cri_rnaseq_2018/cri_rnaseq_2018.star.featurecounts.heatmap.pdf
 </td> </tr> <tr> <th> Dependencies </th> <td> lociStat.featurecounts.star.cri_rnaseq_2018
lociStat.featurecounts.star.cri_rnaseq_2018
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.postAna.pheatmap.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
    <tr> <td> <table class="table"> <tr> <th> Num </th> <td> 49 </td> </tr> <tr> <th> ID </th> <td> task.Submit_cri_rnaseq_2018.postAna.clusterprofiler.featurecounts.star.cri_rnaseq_2018.line_366.id_49 </td> </tr> <tr> <th> Name </th> <td> postAna.clusterprofiler.featurecounts.star.cri_rnaseq_2018 </td></tr> <tr> <th> Thread </th> <td> thread_Root </td> </tr> <tr> <th> PID </th> <td> 10817821.cri16sc001 </td> </tr> </table> </td> <td > <table class="table"> <tr> <th> OK </th> <td> true </td>  </tr> <tr> <th> Exit Code </th> <td> 0 </td>  </tr> <tr> <th> Retries </th> <td>  </td></tr> <tr> <th> State </th> <td> FINISHED </td> </tr> <tr> <th> Dep. </th> <td> OK </td> </tr> <tr> <th> Cpus </th> <td>  4 </td></tr> <tr> <th> Mem </th> <td> 32.0 GB </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Start </th> <td> 2018-11-14 10:58:22 </td> </tr> <tr> <th> End </th> <td> 2018-11-14 11:03:18 </td> </tr> <tr> <th> Elapsed </th> <td> 00:04:55 </td> </tr> <tr> <th> Timeout</th> <td> 3 days </td></tr> <tr> <th> Wall Timeout </th><td>  3 days </td></tr> </table> </td> <td> <table class="table"> <tr> <th> Input files</th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/LociStat/featurecounts/star/cri_rnaseq_2018/cri_rnaseq_2018.star.featurecounts.overlap.txt
 </td>  </tr> <tr> <th> Output files </th> <td> /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/PostAna/clusterprofiler/featurecounts/star/cri_rnaseq_2018/cri_rnaseq_2018.star.featurecounts.enrichGO.ALL.txt
 </td> </tr> <tr> <th> Dependencies </th> <td> lociStat.featurecounts.star.cri_rnaseq_2018
lociStat.featurecounts.star.cri_rnaseq_2018
 </td> </tr> </table> </td> <td> <pre>  bash /gpfs/data/bioinformatics/cri_rnaseq_2018/RNAseq/shell_scripts/run.postAna.clusterprofiler.featurecounts.star.cri_rnaseq_2018.sh; sleep 2 </pre> <pre>     </pre> </td> </tr>
	</table>
</div>

<div class="container" id="scope">
	<h2 class="bs-featurette-title"> Global scope </h2>
	<b>Note:</b> Global scope when program finished execution.
	<table class="table table-striped">
	<tr> <th> Type </th><th> Name </th><th> Value </th> </tr>
	<tr> <td> string </td><td> _ </td><td> /apps/software/gcc-6.2.0/bds/0.99999e/bds </td> </tr>
	<tr> <td> string </td><td> _LMFILES_ </td><td> /apps/modulefiles/Core/gcc/6.2.0.lua:/apps/modulefiles/Core/java-jdk/1.8.0_92.lua:/apps/modulefiles/Compiler/gcc/6.2.0/bds/0.99999e.lua </td> </tr>
	<tr> <td> string </td><td> _ModuleTable001_ </td><td> X01vZHVsZVRhYmxlXz17WyJhY3RpdmVTaXplIl09MyxiYXNlTXBhdGhBPXsiL2FwcHMvbW9kdWxlZmlsZXMvTGludXgiLCIvYXBwcy9tb2R1bGVmaWxlcy9Db3JlIiwiL2FwcHMvZGVmYXVsdC9sbW9kL2xtb2QvbW9kdWxlZmlsZXMvQ29yZSIsfSxbImNfcmVidWlsZFRpbWUiXT1mYWxzZSxbImNfc2hvcnRUaW1lIl09ZmFsc2UsZmFtaWx5PXtbImNvbXBpbGVyIl09ImdjYyIsfSxpbmFjdGl2ZT17fSxtVD17YmRzPXtbIkZOIl09Ii9hcHBzL21vZHVsZWZpbGVzL0NvbXBpbGVyL2djYy82LjIuMC9iZHMvMC45OTk5OWUubHVhIixbImRlZmF1bHQiXT0xLFsiZnVsbE5hbWUiXT0iYmRzLzAuOTk5OTllIixbImxvYWRPcmRlciJdPTMscHJvcFQ9e30sWyJzaG9ydCJdPSJiZHMiLFsi </td> </tr>
	<tr> <td> string </td><td> _ModuleTable002_ </td><td> c3RhdHVzIl09ImFjdGl2ZSIsfSxnY2M9e1siRk4iXT0iL2FwcHMvbW9kdWxlZmlsZXMvQ29yZS9nY2MvNi4yLjAubHVhIixbImRlZmF1bHQiXT0wLFsiZnVsbE5hbWUiXT0iZ2NjLzYuMi4wIixbImxvYWRPcmRlciJdPTEscHJvcFQ9e30sWyJzaG9ydCJdPSJnY2MiLFsic3RhdHVzIl09ImFjdGl2ZSIsfSxbImphdmEtamRrIl09e1siRk4iXT0iL2FwcHMvbW9kdWxlZmlsZXMvQ29yZS9qYXZhLWpkay8xLjguMF85Mi5sdWEiLFsiZGVmYXVsdCJdPTAsWyJmdWxsTmFtZSJdPSJqYXZhLWpkay8xLjguMF85MiIsWyJsb2FkT3JkZXIiXT0yLHByb3BUPXt9LFsic2hvcnQiXT0iamF2YS1qZGsiLFsic3RhdHVzIl09ImFjdGl2ZSIsfSx9LG1wYXRoQT17Ii9hcHBzL21vZHVsZWZpbGVz </td> </tr>
	<tr> <td> string </td><td> _ModuleTable003_ </td><td> L0NvbXBpbGVyL2phdmEtamRrLzEuOC4wXzkyIiwiL2FwcHMvbW9kdWxlZmlsZXMvQ29tcGlsZXIvZ2NjLzYuMi4wIiwiL2FwcHMvbW9kdWxlZmlsZXMvTGludXgiLCIvYXBwcy9tb2R1bGVmaWxlcy9Db3JlIiwiL2FwcHMvZGVmYXVsdC9sbW9kL2xtb2QvbW9kdWxlZmlsZXMvQ29yZSIsfSxbInN5c3RlbUJhc2VNUEFUSCJdPSIvYXBwcy9tb2R1bGVmaWxlcy9MaW51eDovYXBwcy9tb2R1bGVmaWxlcy9Db3JlOi9hcHBzL2RlZmF1bHQvbG1vZC9sbW9kL21vZHVsZWZpbGVzL0NvcmUiLFsidmVyc2lvbiJdPTIsfQ== </td> </tr>
	<tr> <td> string </td><td> _ModuleTable_Sz_ </td><td> 3 </td> </tr>
	<tr> <td> bool </td><td> allowEmpty </td><td> false </td> </tr>
	<tr> <td> string[] </td><td> args </td><td> [] </td> </tr>
	<tr> <td> string </td><td> BASH_ENV </td><td> /apps/default/lmod/lmod/init/bash </td> </tr>
	<tr> <td> string </td><td> BASH_FUNC_ml() </td><td> () {  eval $($LMOD_DIR/ml_cmd \"$@\")\n} </td> </tr>
	<tr> <td> string </td><td> BASH_FUNC_module() </td><td> () {  eval $($LMOD_CMD bash \"$@\");\n [ $? = 0 ] && eval $(${LMOD_SETTARG_CMD:-:} -s sh)\n} </td> </tr>
	<tr> <td> bool </td><td> canFail </td><td> false </td> </tr>
	<tr> <td> string </td><td> CC </td><td> gcc </td> </tr>
	<tr> <td> string </td><td> CFLAGS </td><td> -I/home/wchan10/.local/include </td> </tr>
	<tr> <td> string </td><td> CLASSPATH </td><td> /apps/software/gcc-6.2.0/ngs/1.3.0/jar/ngs-java.jar: </td> </tr>
	<tr> <td> string </td><td> CLICOLOR </td><td> 1 </td> </tr>
	<tr> <td> string </td><td> CPATH </td><td> /apps/default/include:: </td> </tr>
	<tr> <td> string </td><td> CPP </td><td> gcc -E </td> </tr>
	<tr> <td> int </td><td> cpus </td><td> 1 </td> </tr>
	<tr> <td> int </td><td> cpusLocal </td><td> 28 </td> </tr>
	<tr> <td> string </td><td> CVS_RSH </td><td> ssh </td> </tr>
	<tr> <td> string </td><td> CXX </td><td> g++ </td> </tr>
	<tr> <td> int </td><td> day </td><td> 86400 </td> </tr>
	<tr> <td> string </td><td> DISPLAY </td><td> localhost:13.0 </td> </tr>
	<tr> <td> real </td><td> E </td><td> 2.718281828459045 </td> </tr>
	<tr> <td> string </td><td> F77 </td><td> gfortran </td> </tr>
	<tr> <td> string </td><td> F90 </td><td> gfortran </td> </tr>
	<tr> <td> string </td><td> FC </td><td> gfortran </td> </tr>
	<tr> <td> int </td><td> G </td><td> 1073741824 </td> </tr>
	<tr> <td> string </td><td> G_BROKEN_FILENAMES </td><td> 1 </td> </tr>
	<tr> <td> string </td><td> HISTCONTROL </td><td> ignoredups </td> </tr>
	<tr> <td> string </td><td> HISTFILE </td><td> /home/wchan10/.bash_history </td> </tr>
	<tr> <td> string </td><td> HISTFILESIZE </td><td> 5000 </td> </tr>
	<tr> <td> string </td><td> HISTSIZE </td><td> 5000 </td> </tr>
	<tr> <td> string </td><td> HISTTIMEFORMAT </td><td> %b %d %Y %T  </td> </tr>
	<tr> <td> string </td><td> HOME </td><td> /home/wchan10 </td> </tr>
	<tr> <td> string </td><td> HOSTNAME </td><td> cri16in001 </td> </tr>
	<tr> <td> int </td><td> hour </td><td> 3600 </td> </tr>
	<tr> <td> string </td><td> JAVA_HOME </td><td> /apps/compilers/java-jdk/1.8.0_92 </td> </tr>
	<tr> <td> int </td><td> K </td><td> 1024 </td> </tr>
	<tr> <td> string </td><td> LABSHARE </td><td> /gpfs/data/bioinformatics </td> </tr>
	<tr> <td> string </td><td> LANG </td><td> en_US.UTF-8 </td> </tr>
	<tr> <td> string </td><td> LC_ALL </td><td> C </td> </tr>
	<tr> <td> string </td><td> LD_LIBRARY_PATH </td><td> /apps/compilers/gcc/6.2.0/lib/gcc/x86_64-unknown-linux-gnu/6.2.0:/apps/compilers/gcc/6.2.0/lib64:/apps/compilers/gcc/6.2.0/lib:/home/wchan10/.local/lib64:/home/wchan10/.local/lib:/apps/software/gcc-6.2.0/ngs/1.3.0/lib64:/apps/software/gcc-6.2.0/ncbi-vdb/2.8.0/lib64:/apps/default/lib:: </td> </tr>
	<tr> <td> string </td><td> LDFLAGS </td><td> -L/home/wchan10/.local/lib </td> </tr>
	<tr> <td> string </td><td> LESSOPEN </td><td> ||/usr/bin/lesspipe.sh %s </td> </tr>
	<tr> <td> string </td><td> LMOD_arch </td><td> x86_64 </td> </tr>
	<tr> <td> string </td><td> LMOD_CMD </td><td> /apps/default/lmod/lmod/libexec/lmod </td> </tr>
	<tr> <td> string </td><td> LMOD_COLORIZE </td><td> yes </td> </tr>
	<tr> <td> string </td><td> LMOD_DEFAULT_MODULEPATH </td><td> /apps/modulefiles/Linux:/apps/modulefiles/Core:/apps/default/lmod/lmod/modulefiles/Core </td> </tr>
	<tr> <td> string </td><td> LMOD_DIR </td><td> /apps/default/lmod/lmod/libexec </td> </tr>
	<tr> <td> string </td><td> LMOD_FAMILY_COMPILER </td><td> gcc </td> </tr>
	<tr> <td> string </td><td> LMOD_FULL_SETTARG_SUPPORT </td><td> no </td> </tr>
	<tr> <td> string </td><td> LMOD_PKG </td><td> /apps/default/lmod/lmod </td> </tr>
	<tr> <td> string </td><td> LMOD_PREPEND_BLOCK </td><td> normal </td> </tr>
	<tr> <td> string </td><td> LMOD_SETTARG_CMD </td><td> : </td> </tr>
	<tr> <td> string </td><td> LMOD_sys </td><td> Linux </td> </tr>
	<tr> <td> string </td><td> LMOD_VERSION </td><td> 6.4 </td> </tr>
	<tr> <td> string </td><td> LOADEDMODULES </td><td> gcc/6.2.0:java-jdk/1.8.0_92:bds/0.99999e </td> </tr>
	<tr> <td> string </td><td> LOGNAME </td><td> wchan10 </td> </tr>
	<tr> <td> string </td><td> LS_COLORS </td><td> rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;05;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arj=01;31:*.taz=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lz=01;31:*.xz=01;31:*.bz2=01;31:*.tbz=01;31:*.tbz2=01;31:*.bz=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.rar=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.axv=01;35:*.anx=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.axa=01;36:*.oga=01;36:*.spx=01;36:*.xspf=01;36: </td> </tr>
	<tr> <td> string </td><td> LSCOLORS </td><td> ExFxBxDxCxegedabagacad </td> </tr>
	<tr> <td> int </td><td> M </td><td> 1048576 </td> </tr>
	<tr> <td> string </td><td> MAIL </td><td> /var/spool/mail/wchan10 </td> </tr>
	<tr> <td> string </td><td> MAILCHECK </td><td> 300 </td> </tr>
	<tr> <td> string </td><td> MANPATH </td><td> /apps/compilers/java-jdk/1.8.0_92/man:/apps/compilers/gcc/6.2.0/share/man:/home/wchan10/perl5/man:/apps/default/lmod/lmod/share/man:/opt/moab/share/man:/apps/default/share/man:/apps/default/man:: </td> </tr>
	<tr> <td> int </td><td> mem </td><td> -1 </td> </tr>
	<tr> <td> int </td><td> minute </td><td> 60 </td> </tr>
	<tr> <td> string </td><td> MOABHOMEDIR </td><td> /opt/moab </td> </tr>
	<tr> <td> string </td><td> MODULEPATH </td><td> /apps/modulefiles/Compiler/java-jdk/1.8.0_92:/apps/modulefiles/Compiler/gcc/6.2.0:/apps/modulefiles/Linux:/apps/modulefiles/Core:/apps/default/lmod/lmod/modulefiles/Core </td> </tr>
	<tr> <td> string </td><td> MODULEPATH_ROOT </td><td> /apps/modulefiles </td> </tr>
	<tr> <td> string </td><td> MODULESHOME </td><td> /apps/default/lmod/lmod </td> </tr>
	<tr> <td> string </td><td> MSM_HOME </td><td> /usr/local/MegaRAID Storage Manager </td> </tr>
	<tr> <td> string </td><td> MSM_PRODUCT </td><td> MSM </td> </tr>
	<tr> <td> string </td><td> NGS_LIBDIR </td><td> /apps/software/gcc-6.2.0/ngs/1.3.0/lib64 </td> </tr>
	<tr> <td> string </td><td> NLSPATH </td><td> /usr/dt/lib/nls/msg/%L/%N.cat </td> </tr>
	<tr> <td> string </td><td> node </td><td>  </td> </tr>
	<tr> <td> int </td><td> P </td><td> 1125899906842624 </td> </tr>
	<tr> <td> string </td><td> PATH </td><td> /apps/software/gcc-6.2.0/bds/0.99999e/mesos:/apps/software/gcc-6.2.0/bds/0.99999e:/apps/compilers/java-jdk/1.8.0_92/bin:/apps/compilers/gcc/6.2.0/bin:/home/wchan10/perl5/bin:/home/wchan10/.local/bin:/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/BigDataScript/.bds:/gpfs/data/bioinformatics/wchan10/TOOLS/MEME/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/snpEff/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/vt:/gpfs/data/bioinformatics/wchan10/TOOLS/HTSlib/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/vcflib/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/vcfsorter:/gpfs/data/bioinformatics/wchan10/TOOLS/VCFtools/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/seqtk:/gpfs/data/bioinformatics/wchan10/TOOLS/UCSCKentUtils:/gpfs/data/bioinformatics/wchan10/TOOLS/UCSCKentUtils/blat:/gpfs/data/bioinformatics/wchan10/TOOLS/sambamba:/gpfs/data/bioinformatics/wchan10/TOOLS/IGVTools/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/bedtools/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/BEDOPS/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/GitHub/picard/build/libs:/gpfs/data/bioinformatics/wchan10/TOOLS/bam-readcount/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/BCFtools/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/SAMStat/CURRENT/src:/gpfs/data/bioinformatics/wchan10/TOOLS/methylQA:/gpfs/data/bioinformatics/wchan10/TOOLS/GATK:/gpfs/data/bioinformatics/wchan10/TOOLS/freebayes/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/kallisto/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/Novocraft/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/BWA/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/TopHat/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/Bowtie2/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/FLASH/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/Trimmomatic/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/SeqPrep:/gpfs/data/bioinformatics/wchan10/TOOLS/FastQC/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/pigz/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/CMake/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/NCBIDecryptionTools/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/sratoolkit/CURRENT/bin:/gpfs/data/bioinformatics/wchan10/TOOLS/PLINK/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/libSVM/CURRENT:/gpfs/data/bioinformatics/wchan10/TOOLS/libSVM/CURRENT/tools:/usr/local/bin:/usr/local/sbin:/apps/software/gcc-6.2.0/sra-tools/2.8.0/bin:/usr/lib64/qt-3.3/bin:/opt/moab/bin:/apps/default/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/cuda/bin:/opt/ibutils/bin:/opt/phoenix/bin </td> </tr>
	<tr> <td> string </td><td> PERL5LIB </td><td> /home/wchan10/perl5/lib/perl5:/gpfs/data/bioinformatics/wchan10/TOOLS/VCFtools/CURRENT/src/perl:/opt/moab/lib/perl5 </td> </tr>
	<tr> <td> string </td><td> PERL_LOCAL_LIB_ROOT </td><td> /home/wchan10/perl5 </td> </tr>
	<tr> <td> string </td><td> PERL_MB_OPT </td><td> --install_base \"/home/wchan10/perl5\" </td> </tr>
	<tr> <td> string </td><td> PERL_MM_OPT </td><td> INSTALL_BASE=/home/wchan10/perl5 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_ARCH </td><td> x86_64 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_BASENAME_STARTNUM </td><td> 1 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_DISTRODIR </td><td> /opt/phoenix/install/RHEL/6.7/x86_64 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_ETCDIR </td><td> /opt/phoenix/etc/phoenix </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_DHCPD_KICK_END </td><td> 10.50.85.254 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_DHCPD_KICK_START </td><td> 10.50.85.1 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_HOSTNAME </td><td> cri16sc001 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IB </td><td> y </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IB_IP </td><td> 10.50.86.252 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IB_NETMASK </td><td> 255.255.254.0 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IB_SUBNET </td><td> 10.50.86.0 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IPMI </td><td> y </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IPMI_GATEWAY </td><td> 10.50.88.140 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IPMI_IP </td><td> 10.50.88.252 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IPMI_NETMASK </td><td> 255.255.254.0 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_IPMI_SUBNET </td><td> 10.50.88.0 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_PRIVATE_IP </td><td> 10.50.84.252 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_PRIVATE_MTU </td><td> 9000 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_PRIVATE_NETMASK </td><td> 255.255.254.0 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_PRIVATE_SUBNET </td><td> 10.50.84.0 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_HEAD_TIMEZONE </td><td> America/Chicago </td> </tr>
	<tr> <td> string </td><td> PHOENIX_LINUX </td><td> RHEL </td> </tr>
	<tr> <td> string </td><td> PHOENIX_LINUX_VERSION </td><td> 6.7 </td> </tr>
	<tr> <td> string </td><td> PHOENIX_NFS_SHAREDIR </td><td> /shared </td> </tr>
	<tr> <td> string </td><td> PHOENIX_ROOTDIR </td><td> /opt/phoenix </td> </tr>
	<tr> <td> string </td><td> PHOENIX_VERSION </td><td> 2.1.0 </td> </tr>
	<tr> <td> real </td><td> PI </td><td> 3.141592653589793 </td> </tr>
	<tr> <td> string </td><td> PKG_CONFIG_PATH </td><td> /home/wchan10/.local/lib/pkgconfig: </td> </tr>
	<tr> <td> string </td><td> ppwd </td><td> /gpfs/data/bioinformatics/cri_rnaseq_2018 </td> </tr>
	<tr> <td> string </td><td> programName </td><td> Submit_cri_rnaseq_2018.bds </td> </tr>
	<tr> <td> string </td><td> programPath </td><td> /gpfs/data/bioinformatics/cri_rnaseq_2018/Submit_cri_rnaseq_2018.bds </td> </tr>
	<tr> <td> string </td><td> PROMPT_COMMAND </td><td> history -a </td> </tr>
	<tr> <td> string </td><td> PWD </td><td> /gpfs/data/bioinformatics/cri_rnaseq_2018 </td> </tr>
	<tr> <td> string </td><td> QTDIR </td><td> /usr/lib64/qt-3.3 </td> </tr>
	<tr> <td> string </td><td> QTINC </td><td> /usr/lib64/qt-3.3/include </td> </tr>
	<tr> <td> string </td><td> QTLIB </td><td> /usr/lib64/qt-3.3/lib </td> </tr>
	<tr> <td> string </td><td> queue </td><td>  </td> </tr>
	<tr> <td> int </td><td> retry </td><td> 0 </td> </tr>
	<tr> <td> string </td><td> SHELL </td><td> /bin/bash </td> </tr>
	<tr> <td> string </td><td> SHLVL </td><td> 2 </td> </tr>
	<tr> <td> string </td><td> SSH_ASKPASS </td><td> /usr/libexec/openssh/gnome-ssh-askpass </td> </tr>
	<tr> <td> string </td><td> SSH_CLIENT </td><td> 128.135.197.192 55101 22 </td> </tr>
	<tr> <td> string </td><td> SSH_CONNECTION </td><td> 128.135.197.192 55101 10.50.178.250 22 </td> </tr>
	<tr> <td> string </td><td> SSH_TTY </td><td> /dev/pts/297 </td> </tr>
	<tr> <td> string </td><td> system </td><td> cluster </td> </tr>
	<tr> <td> int </td><td> T </td><td> 1099511627776 </td> </tr>
	<tr> <td> string </td><td> TACC_FAMILY_COMPILER </td><td> gcc </td> </tr>
	<tr> <td> string </td><td> TERM </td><td> xterm-color </td> </tr>
	<tr> <td> string </td><td> TIMEFORMAT </td><td> \nWall Time\t%3lR\nUser Mode\t%3lU\nKernel Mode\t%3lS\nCPU Usage\t%P% </td> </tr>
	<tr> <td> int </td><td> timeout </td><td> 86400 </td> </tr>
	<tr> <td> string </td><td> TMPDIR </td><td> /gpfs/data/bioinformatics/tmp </td> </tr>
	<tr> <td> string </td><td> USER </td><td> wchan10 </td> </tr>
	<tr> <td> int </td><td> walltimeout </td><td> 86400 </td> </tr>
	<tr> <td> string </td><td> WCTOOLS </td><td> /gpfs/data/bioinformatics/wchan10/TOOLS </td> </tr>
	<tr> <td> int </td><td> week </td><td> 604800 </td> </tr>
	<tr> <td> string </td><td> XFILESEARCHPATH </td><td> /usr/dt/app-defaults/%L/Dt </td> </tr>
	</table>
</div>

</body>

